### DISCUSSION TITLE
Above 4G Decoding setting missing

### DISCUSSION THREAD

**Discussion by itasli** [terminatorul/NvStrapsReBar#20] [Q&A] [answered]
Hey there, I'm on laptop (don't know if anyone already tested this on laptop btw) and i don't see Above 4G Decoding option at all... does that mean I can't use this tool ?

My computer model is : GL65 Leopard 10SFSK

**Comment by terminatorul** [comment, marked as answer]
Follow the instructions from [ReBarUEFI](https://github.com/xCuri0/ReBarUEFI). Can you find the option if you follow this steps from here:
      https://github.com/xCuri0/ReBarUEFI/wiki/Enabling-hidden-4G-decoding
?

Also can you open Device Manager, select View Resources by Type, and check if you have memory resources above 4 GiB ?

Do you know how to clear CMOS and can you check that it will reset current year to a value before 2024 ? If so, you can reflash your motherboard to add NvStrapsReBar.ffs, then enable ReBAR using NvStrapsReBar.exe, and see for yourself if it works...

**Reply by terminatorul** [reply]
Even without Above 4G Decoding, you should still benefit from a BAR size of 1 GiB or 2 GiB

**Reply by itasli** [reply]
Hey there, thanks for your reply.

After searching for "4G Decod" I did get something : 
![image](https://github.com/terminatorul/NvStrapsReBar/assets/50218050/d54fc9e2-8837-425c-8148-ef2c3cb8aadd)

I don't understand where I can find memory resources info on Device Manager :
![image](https://github.com/terminatorul/NvStrapsReBar/assets/50218050/3f9653c3-2833-4b30-b9b2-9753493a36af)

Clearing CMOS should be "okayish" i guess

**Reply by terminatorul** [reply]
![image](https://github.com/terminatorul/NvStrapsReBar/assets/378924/b477f304-f665-4c9b-befc-9e8187a406a8)

You should follow through with the instructions to enable the hidden option

**Reply by terminatorul** [reply]
Try clearing CMOS ... I guess for a laptop you just pull out the battery for 5 min ... you may need a small screwdriver.
After that, enter UEFI setup on first boot and read the current year. The value should be before 2024

**Reply by itasli** [reply]
Ok got this:
![image](https://github.com/terminatorul/NvStrapsReBar/assets/50218050/8c8fa280-51d8-473a-a975-73bb5fa5c47a)

> Try clearing CMOS ... I guess for a laptop you just pull out the battery for 5 min ... you may need a small screwdriver. After that, enter UEFI setup on first boot and read the current year. The value should be before 2024

Yup will try this asap and let you know ! I watched some disassembly video for it

**Reply by terminatorul** [reply]
Is there no section for Large Memory as well ?
![image](https://github.com/terminatorul/NvStrapsReBar/assets/378924/71ef4d7d-c880-499b-b24a-d3f44e163568)

So is it necessary to open the laptop case to clear CMOS ? What if you remove the battery only ?

**Reply by itasli** [reply]
No Large Memory section :
![image](https://github.com/terminatorul/NvStrapsReBar/assets/50218050/67bb9ec1-3de2-48d3-a2d0-c01b1a26fd9d)

I guess this mean i can/should/need enable Above 4G Decoding using this : [https://github.com/xCuri0/ReBarUEFI/wiki/Enabling-hidden-4G-decoding](https://github.com/xCuri0/ReBarUEFI/wiki/Enabling-hidden-4G-decoding) ?

> So is it necessary to open the laptop case to clear CMOS ? What if you remove the battery only ?

Yes need to remove some screws in ordrer to access CMOS battery, I can't/don't want to reset right now as I had undervolted some times ago and don't how to do it again once reseted !

**Reply by terminatorul** [reply]
Yes, now you need to enable Above 4G Decoding ... or stick with a 1 GiB BAR

**Reply by itasli** [reply]
> Yes, now you need to enable Above 4G Decoding ... or stick with a 1 GiB BAR

I see thanks again for both your work & help ! going to set the discussion as resolved hope it can help other laptop users too

**Comment by terminatorul** [comment]
@itasli 
>  (don't know if anyone already tested this on laptop btw)

There are some laptop systems in the list of working cards at https://github.com/terminatorul/NvStrapsReBar/issues/1

**Comment by itasli** [comment]
Hey @terminatorul I did follow the guide to enable Above 4G decoding
![photo1708376206](https://github.com/terminatorul/NvStrapsReBar/assets/50218050/3effbc13-60a8-484e-bb82-e6cf1cafc427)

 but when I check on GPU-Z i have :

 `Above 4G Decode enabled in BIOS	 No`
 
 I tried both :
  `4G Decod` got offset `0x3` and set to 0x1 using setup_var (Above 4G Decoding in extracted txt file)
  `Above 4G` got offset `0xFA` and set to 0x1 using setup_var_3 (setup_var wasn't working for some reason : out of range or something like that) (Above 4GB MMIO BIOS assignment in extracted txt file)
  I tried last one `MMIO` extracted body but it wasn't recognized by IRFExtractor

**Comment by itasli** [comment]
Ok, I managed to get Above 4G Decode enabled. For MSI Laptop user you my want to check in System Agen (SA) Configuration in Advanced Menu (after enabling advanced settings).

![image](https://github.com/terminatorul/NvStrapsReBar/assets/50218050/6da7fe4d-346a-431f-adcc-ecf25ca34cae)

![image](https://github.com/terminatorul/NvStrapsReBar/assets/50218050/17a72a0d-eeb5-4348-bca8-5a132b6c4702)

Seems like everything is aligned for me to flash and enable ReBar

**Reply by terminatorul** [reply]
Remember to enable Above 4G Decoding again after you flash the motherboard, and before you enable ReBAR

**Reply by mecton1** [reply]
Hi, I have the same MSI GL65 Leopard 10SFSK with an RTX 2070 Super. I enabled Above 4GB MMIO BIOS assignment, but ReBAR still shows disabled in GPU-Z.

Did you successfully flash the modified BIOS and enable ReBAR on this model? If so, which flashing method did you use, and did you need an SPI programmer or Intel FPT? Any details would be greatly appreciated.
<img width="378" height="536" alt="image" src="https://github.com/user-attachments/assets/66876306-1b90-4180-a853-81ead2e1e2e2" />
